### PR TITLE
switch from specific attribute precedence to whatever chef resolves to

### LINKED ISSUE #3: why use normal attributes?
I am not sure I understand why there is a requirement that these are normal attributes which persist chef client runs. I think during the guard it should be `node` as opposed to `normal` as it should resolve to whatever attribute precedence level is highest. This leave it to the user to decide whether they want to use normal, default, etc. Many AWS envs have ephemeral instances that come up and down on their own this makes it harder to use this cookbook.

### PR DESCRIPTION
**Pull request by majormoses**
Fixes #3 

Also misc other styalistic (largely based on rubocop rulesets) changes.

Signed-off-by: Ben Abrams <me@benabrams.it>

### CODE REVIEW

**Comment by majormoses** [comment]
I have not yet tested this and I will have some time to test this tomorrow, the code should be sane though.

**Comment by majormoses** [comment]
Hmm I guess it is a breaking change unless we change the default to false...let me know if you want me to change that.

**Comment by majormoses** [comment]
I tested this and it works as expected, will await to hear back if you want me to change the default to false for inspector being enabled which keeps the backwards compatibility.

**Comment by majormoses** [comment]
I did not see any integration testing framework such as `(server|in)spec` in place or I would have written a test for it but I can verify it installed:
```
babrams@ip-10-55-142-7:~$ dpkg --get-selections | grep awsagent
awsagent					install
awsagentkernelmodule-4.4.0-1047-aws		install
```

File: recipes/default.rb [hunk 1]
```diff
@@ -68,9 +68,9 @@
 remote_file "#{Chef::Config[:file_cache_path]}/install.sig" do
   source              node.default[:inspector][:gpg_signature_url]
   use_conditional_get true
-  mode                0440
+  mode                0o440
```
Comment (eternaltyro) [inline recipes/default.rb:71]: What's this change?

Comment (eternaltyro) [reply recipes/default.rb:71]: Does the 'o' mean octal? Here's what's in the document https://docs.chef.io/resource_directory.html

> UNIX- and Linux-based systems: A quoted 3-5 character string that defines the octal mode that is passed to chmod. For example: '755', '0755', or 00755. If the value is specified as a quoted string, it works exactly as if the chmod command was passed. If the value is specified as an integer, prepend a zero (0) to the value to ensure that it is interpreted as an octal number. For example, to assign read, write, and execute rights for all users, use '0777' or '777'; for the same rights, plus the sticky bit, use 01777 or '1777'.

**Review by eternaltyro — COMMENTED** [review]
Please clarify mode changes.

File: recipes/default.rb (hunk 1 above)
Comment (majormoses) [reply recipes/default.rb:71]: yes it stands for octal

https://github.com/bbatsov/rubocop/blob/master/manual/cops_style.md#stylenumericliteralprefix

Comment (majormoses) [reply recipes/default.rb:71]: I can remove it if you want just a stylistic change, I probably spend too much time writing ruby so I spot these kinds of things but the original is still valid. One is explicit and the other implicit

Comment (eternaltyro) [reply recipes/default.rb:71]: No, IMO explicit is better. Always.

**Comment by eternaltyro** [comment]
Merged. Thanks.

**Comment by majormoses** [comment]
Awesome, if possible please let me know when you bump the version and release the new version to supermarket as I'd love to start using this.

**Comment by eternaltyro** [comment]
Cookbook uploaded. Added you as a collaborator on Supermarket too! Thanks a ton for the help.

**Comment by majormoses** [comment]
@eternaltyro if you are gonna add me as a collaborator on the supermarket could you add me as collaborator on github?

**Comment by eternaltyro** [comment]
@majormoses also done.

**Pull request merged**